### ISSUE TITLE
No dimensionality checking is done for LocalEnergyMinimizer.minimize tolerance

### ISSUE THREAD

**Issue by jchodera** [openmm/openmm#4022]
Currently, it's possible to call `LocalEnergyMinimizer.minimize()` with a tolerance that has any dimensionality:
```python
>>> import openmm
>>> from openmmtools.testsystems import WaterBox
>>> t = WaterBox()
>>> integrator = openmm.VerletIntegrator(0.001)
>>> context = openmm.Context(t.system, integrator)
>>> context.setPositions(t.positions)
>>> from openmm import unit
>>> openmm.LocalEnergyMinimizer.minimize(context, 1.0*unit.kilojoules_per_mole)
>>> openmm.LocalEnergyMinimizer.minimize(context, 1.0*unit.kilojoules_per_mole/unit.nanometers)
>>>
```
Could we add [dimensionality checking in this block](https://github.com/openmm/openmm/blob/master/wrappers/python/src/swig_doxygen/swigInputConfig.py#L159) to `minimize`?

**Comment by zhang-ivy** [comment]
Could we also report the units for tolerance in the docs? Currently it just says:
```
tolerance (double) – this specifies how precisely the energy minimum must be located. 
Minimization will be halted once the root-mean-square value of all force components reaches this tolerance. 
The default value is 10.
```

**Comment by peastman** [comment]
The fix is in #4035.

**Comment by jchodera** [comment]
Thanks!
